### PR TITLE
feat: implement a debug setting for the client

### PR DESCRIPTION
**Pull request by sillyElvy**
## About

This change makes it easier to enable debug logging and gives debug logging.

## Checklist

- [x] The ``pre-commit`` code linter has been run over all edited files to ensure the code is linted.
- [x] I've ensured the change(s) work on `3.8.6` and higher.

I've made this pull request: (check all that apply)
  - [ ] For the documentation
  - [x] To add a new feature
  - [x] As a general enhancement
  - [ ] As a refactor of the library/the library's code
  - [ ] To fix an existing bug
  - [ ] To resolve #ISSUENUMBER

This is:
  - [ ] A breaking change

### CODE REVIEW

**Review by Toricane — COMMENTED** [review]
Should there be a way to change the level to stuff other than debug?

**Comment by sillyElvy** [comment]
> Should there be a way to change the level to stuff other than debug?

Idk, if it'd be wanted I can bring back the previous commit but modify it a bit

**Comment by Toricane** [comment]
Maybe name it `logging` or something instead of `debug`, where you can input a logging level or bool which defaults to debug or off

**Comment by sillyElvy** [comment]
> Maybe name it `logging` or something instead of `debug`, where you can input a logging level or bool which defaults to debug or off

Done

File: interactions/client/bot.py [hunk 1]
```diff
@@ -102,6 +104,21 @@ def __init__(
...
+            # thx i0 for posting this on the retux Discord
+
+            if _logging is True:
+                _logging = logging.DEBUG
+
+            _format = (
+                "%(asctime)s [%(levelname)s] - .%(funcName)s(): %(message)s"
+                if _logging == logging.DEBUG
+                else "%(asctime)s [%(levelname)s] - %(message)s"
+            )
```
Comment (Toricane) [inline interactions/client/bot.py:118]: Why is the format different? Is funcName given only in DEBUG level?

Comment (sillyElvy) [reply interactions/client/bot.py:118]: I don't think so - but why would you want the func name if you are not debugging

Comment (Toricane) [reply interactions/client/bot.py:118]: Maybe to know where it errored or warned? Idk lmk if it's really unnecessary and ill approve

Comment (sillyElvy) [reply interactions/client/bot.py:118]: I think it's unnecessary

**Review by Toricane — APPROVED** [review]
LGTM!

**Pull request merged**